testsuite/integration/manualmode/src/test/java/org/wildfly/test/manual/elytron/seccontext/ElytronClientDefaultSSLContextProviderTestCase.java

```diff
@@ -0,0 +1,95 @@
+package org.wildfly.test.manual.elytron.seccontext;
+
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.wildfly.security.auth.client.ElytronClientDefaultSSLContextProvider;
+import org.wildfly.security.x500.cert.BasicConstraintsExtension;
+import org.wildfly.security.x500.cert.X509CertificateBuilder;
+
+import javax.net.ssl.SSLContext;
+import javax.security.auth.x500.X500Principal;
+import javax.ws.rs.client.ClientBuilder;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.Security;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+public class ElytronClientDefaultSSLContextProviderTestCase {
+
+    private static final String CONFIG_FILE = "file:./src/test/resources/wildfly-config-default-ssl-context.xml";
+    private static final String CLIENT_KEYSTORE = "client.keystore";
+    private static final char[] PASSWORD = "password".toCharArray();
+    private static final String ALIAS = "alias";
+    private static File KEYSTORES_DIR = new File("./target/keystores");
+
+    @BeforeClass
+    public static void generateKeystoreAndConfigureProvider() throws CertificateException, NoSuchAlgorithmException, KeyStoreException, IOException {
+        new File(KEYSTORES_DIR, CLIENT_KEYSTORE).delete();
+        KEYSTORES_DIR.delete();
+        generateClientKeystore();
+        Security.insertProviderAt(new ElytronClientDefaultSSLContextProvider(CONFIG_FILE), 1);
+    }
+
+    @Test
+    public void testDefaultSSLContextIsNotReturnedIfNotConfigured() {
+        Assert.assertNotNull(Security.getProvider(ElytronClientDefaultSSLContextProvider.class.getSimpleName()));
+        try {
+            ResteasyClientBuilder builder = (ResteasyClientBuilder) ClientBuilder.newBuilder();
+            ResteasyClient resteasyClient = builder.sslContext(SSLContext.getDefault()).build();
+            Assert.assertEquals(resteasyClient.getSslContext().getProvider().getName(), ElytronClientDefaultSSLContextProvider.class.getSimpleName());
+        } catch (NoSuchAlgorithmException e) {
+            Assert.fail();
+        }
+    }
+
+    private static void generateClientKeystore() throws KeyStoreException, CertificateException, NoSuchAlgorithmException, IOException {
+        if (!KEYSTORES_DIR.exists()) {
```

Comment:
This whole block seems a bit redundant, since you clean up the keystore filesystem before and after the test, but I understand this is because it could happen that there is some leftover, correct? 
E.g.: when testing locally, I guess.

In such a case then it could happen that there's still the chance that a `File(KEYSTORES_DIR, CLIENT_KEYSTORE)` exists, right? I such case.... should we check for it to prevent IOException whren it's there already?

let me know in case you think I misunderstood the test life cycle.

Reply:
@fabiobrz I removed this block from @BeforeClass method. It is sufficient IMO to create this directory before generation of keystore, and then remove it only in the @AfterClass  method.

Reply:
Not sure exactly where you want to check for IOException, bu maybe this was addressed by my removing this block from `BeforeClass` method?

Reply:
> I removed this block from @BeforeClass method. It is sufficient IMO to create this directory before generation of keystore, and then remove it only in the @AfterClass method.

IMO it is, when the test is executed in a clean environment, although you might need the removal to be executed also preemptively before the test in local environments, when for instance the previous execution didn't reach the cleanup method for some reason.

Reply:
> Not sure exactly where you want to check for IOException, bu maybe this was addressed by my removing this block from BeforeClass method?

Sorry, maybe wrong wording from my side.
I didn't mean we need to check for IOException. What I meant is that the key store filesystem is cleaned up in pre and post test methods. So it should not exist.

Then, in this method, you check again that it does not exist before creating it, which means that you allow for some cases in which it could happen that it exist, so this is why I said it seems redundant but... _better safe than sorry_
So I am okay for this block to be there.

Regarding the IOException, ignore my concern, I though you were creating the file again, while it is just written, which fits in  with the case in which the directory survived some exection.

Reply:
@fabiobrz Ok so did you mean I should add smth like this to the BeforeClass method? : 

```
if (KEYSTORES_DIR.exists()) {
            new File(KEYSTORES_DIR, CLIENT_KEYSTORE).delete();
            KEYSTORES_DIR.delete();
}
```

Reply:
> Ok so did you mean I should add smth like this to the BeforeClass method? :

Yes, it could be cleaner, IMHO, but I don't have strong opinion to be honest. 

The reason for the initial comment was that it seemed redundant to have those cleanup called pre and post the test execution and then using an existing file in case it's there.

Reply:
@fabiobrz I see now, yes that's true, thank you for explanation. I added the block to the BeforeClass method.